### PR TITLE
Revert "Merge pull request #2778 from nilsleiffischer/fix_pch_flags"

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

The PCH changes broke the build. Marking as priority because I need to be able to build the code.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

**Review by moxcodes — APPROVED** [review]
Yeh. woops.

**Review by nilsvu — APPROVED** [review]

File: cmake/SetupPch.cmake [hunk 1]
```diff
@@ -206,10 +210,11 @@ if (USE_PCH)
     )
 
   # Create the ${SPECTRE_PCH} that libraries and executables will depend on
-  add_library(${SPECTRE_PCH} INTERFACE)
-  target_link_libraries(
+  add_custom_target(
```
Comment (nilsdeppe) [inline cmake/SetupPch.cmake:213]: I think it was this change that broke things, but I don't have time to investigate

**Comment by nilsvu** [comment]
Yep let's revert and I'll look at the issue later this week. Not sure why this wasn't caught on CI though

**Comment by nilsdeppe** [comment]
:+1: my guess is it's either because I use ninja and so the dependency graph is evaluated in a slightly different order, or it's because CI does `-j2` while I do `-j62`

**Comment by nilsdeppe** [comment]
#2782 gets the dependencies correct and still propagates flags, etc. forward. Closing this.

**Pull request closed without merging**